mllam_data_prep/ops/selection.py

```diff
@@ -56,64 +43,44 @@ def select_by_kwargs(ds, **coord_ranges):
     """
 
     for coord, selection in coord_ranges.items():
-        if coord not in ds.coords:
-            raise ValueError(f"Coordinate {coord} not found in dataset")
-        if isinstance(selection, Range):
-            if selection.start is None and selection.end is None:
-                raise ValueError(
-                    f"Selection for coordinate {coord} must have either 'start' and 'end' given"
-                )
-            sel_start = _normalize_slice_startstop(selection.start)
-            sel_end = _normalize_slice_startstop(selection.end)
-            sel_step = _normalize_slice_step(selection.step)
-
-            assert sel_start != sel_end, "Start and end cannot be the same"
-
-            # we don't select with the step size for now, but simply check (below) that
-            # the step size in the data is the same as the requested step size
-            ds = ds.sel({coord: slice(sel_start, sel_end)})
-
-            if coord == "time":
-                check_point_in_dataset(coord, sel_start, ds)
-                check_point_in_dataset(coord, sel_end, ds)
-                if sel_step is not None:
-                    check_step(sel_step, coord, ds)
-
-            assert (
-                len(ds[coord]) > 0
-            ), f"You have selected an empty range {sel_start}:{sel_end} for coordinate {coord}"
-
-        elif isinstance(selection, list):
-            ds = ds.sel({coord: selection})
-        else:
-            raise NotImplementedError(
-                f"Selection for coordinate {coord} must be a list or a dict"
-            )
+        sel_start = selection.start
+        sel_end = selection.end
+        sel_step = selection.step
+
+        if coord == "time":
+            sel_start = to_timestamp(selection.start)
+            sel_end = to_timestamp(selection.end)
+            sel_step = get_time_step(sel_step, ds)
+
+        assert sel_start != sel_end, "Start and end cannot be the same"
+
```

Comment:
`ds.sel({'time': slice(dt.datetime(2023,1,1), dt.datetime(2023,1,1))})` is a valid expression and returns data from `dt.datetime(2023,1,1)`